extensions/notebook/src/book/remoteBookController.ts

```diff
@@ -0,0 +1,146 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as request from 'request';
+import * as loc from '../common/localizedConstants';
+import * as utils from '../common/utils';
+import * as vscode from 'vscode';
+import { RemoteBookDialogModel } from '../dialog/remoteBookDialog';
+import { GitHubRemoteBook } from '../book/githubRemoteBook';
+import { SharedRemoteBook } from '../book/sharedRemoteBook';
+
+const assetNameRE = /([a-zA-Z0-9]+)(?:-|_)([a-zA-Z0-9.]+)(?:-|_)([a-zA-Z0-9]+).(zip|tar.gz|tgz)/;
+
+export class RemoteBookController {
+	constructor(public model: RemoteBookDialogModel, public outputChannel: vscode.OutputChannel) {
+	}
+
+	public async setRemoteBook(url: URL, remoteLocation: string, asset?: IAsset): Promise<void> {
+		if (remoteLocation === 'GitHub') {
+			this.model.remoteBook = new GitHubRemoteBook(url, this.outputChannel, asset);
+		} else {
+			this.model.remoteBook = new SharedRemoteBook(url, this.outputChannel);
+		}
+		return await this.model.remoteBook.createLocalCopy();
+	}
+
+	public async getReleases(url?: URL): Promise<IRelease[]> {
+		if (url) {
+			this.model.releases = [];
+			let options = {
+				headers: {
+					'User-Agent': 'request'
```

Comment:
Is this needed by GitHub or something? Not sure what a User-Agent of request would be - do you have any context on that? What happens if you remove it?

Reply:
It seems that now GitHub requires to set the User-Agent header, if not the request response it's always 403.

Reply:
I think that instead of 'request', it should be the App name, so probably I should modify it to azuredatastudio.

Reply:
Yeah that seems fine. I'll admit I'm not sure what the consequences are of not using a common user agent string but if it accepts it then let's go with that.